include/revng/Model/RevngEnforcers.h

```diff
@@ -0,0 +1,186 @@
+#pragma once
+
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+#include <array>
+
+#include "llvm/IR/LegacyPassManager.h"
+
+#include "revng/Model/AutoEnforcerTarget.h"
+#include "revng/Model/BackingContainers.h"
+#include "revng/Model/BinaryContainer.h"
+#include "revng/Model/InputOutputContract.h"
+#include "revng/Model/LLVMEnforcer.h"
+namespace Model {
+
+extern Granularity RootGranularity;
+extern Granularity FunctionsGranularity;
+
+extern Kind CFepper;
+extern Kind Binary;
+extern Kind Root;
+extern Kind RootIsolated;
+extern Kind Support;
+extern Kind Object;
+extern Kind Translated;
+
+extern Kind Isolated;
+extern Kind ABIEnforced;
+
+extern Kind Dead;
+
+class LiftEnforcer {
+public:
+  static constexpr auto Name = "Lift Enforcer";
+  std::array<InputOutputContract, 1> getContract() const {
+    return { InputOutputContract(Binary, KindExactness::Exact, 0, Root, 1) };
+  }
+
+  void run(const BinaryContainer &SourceBinary,
+           DefaultLLVMContainer &TargetContainer) {
+    revng_abort("Not implemented");
+  }
+};
+
+class CFepperEnforcer {
+public:
+  static constexpr auto Name = "CFepper Enforcer";
+  std::array<InputOutputContract, 1> getContract() const {
+    return {
+      InputOutputContract(Root, KindExactness::Exact, 0, CFepper, true)
+    };
+  }
```

Comment:
Can this be `static constexpr`?
I think you explained me why it can't, but it seems strange now.

Reply:
Since the contract may depend onto member variables it is cannot be static, if needed we can do some introspection magic and look for both the static and non static getContract member function and prefer the non static if there are both, but that would entail less clear compilation errors when none is present.

Reply:
Can you make an example of a situation when you need member variables?
Again, I just don't remember.

Reply:
as an example, the link1 and link2 that require the root or root isolated functions may either be implemented as two different classes or as one class with a configuration taken in the constructor that specify which one to use.